src/util-logopenfile.c

```diff
@@ -83,9 +129,21 @@ static int SCLogFileWrite(const char *buffer, int buffer_len, LogFileCtx *log_ct
 
     int ret = 0;
 
+    if (log_ctx->fp == NULL && log_ctx->is_sock)
+        ret = SCLogUnixSocketReconnect(log_ctx);
```

Comment:
I think it's a bit unclear now what SCLogFileWrite returns: it can either return the retval of SCLogUnixSocketReconnect, or of fwrite.